File: Lib.Repository/Entities/Position.cs
namespace Lib.Repository.Entities
{
    public class Position
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Location { get; set; }
        public int DepartmentId { get; set; }
        public int? RecruiterId { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public decimal? Budget { get; set; }
        public DateTime? ClosingDate { get; set; }
        public string? Status { get; set; }
        public string? RecruiterName { get; set; }
        public string? DepartmentName { get; set; }
    }
}

File: WebApp/src/types/position.ts
import type { Status } from '../constants/positionConstants';

export interface Position {
  id?: string; // Keep for backward compatibility
  positionId?: number | string; // Add positionId to match backend
  title: string;
  description: string;
  location: string;
  status: Status;
  recruiterId: number | string; // Can be number or string for form handling
  departmentId: number | string; // Can be number or string for form handling
  budget: number;
  closingDate?: string | null; // Allow null for form handling
}

export interface PositionTableProps {
  positions: Position[];
  expandedIndex: number | null;
  onExpand: (idx: number) => void;
  onDelete: (id: string) => void;
  onUpdate: (id: string, position: Position) => void;
  onAdd?: (position: Position) => void;
} 